clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -3102,11 +3103,14 @@ void CXXNameMangler::mangleType(QualType T) {
     addSubstitution(T);
 }
 
-void CXXNameMangler::mangleCXXRecordDecl(const CXXRecordDecl *Record) {
+void CXXNameMangler::mangleCXXRecordDecl(const CXXRecordDecl *Record,
+                                         bool DontAddSubstitutionForCompat) {
   if (mangleSubstitution(Record))
     return;
   mangleName(Record);
-  if (isCompatibleWith(LangOptions::ClangABI::Ver19))
+  // Return early without adding the record to the dictionary of substitution
+  // candidates to maintain compatibility with older ABIs.
```

Comment:
This comment probably belongs on the call site. I would also suggest dropping `ForCompat` from the parameter name.